src/app/interceptors/global-http-error.interceptor.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable, ErrorHandler } from '@angular/core';
+import {
+  HttpInterceptor,
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpErrorResponse
+} from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+@Injectable()
+export class GlobalHttpErrorInterceptor implements HttpInterceptor {
+  constructor(private errorHandler: ErrorHandler) {}
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return next.handle(req).pipe(
+      catchError((error) => {
+        if (error instanceof HttpErrorResponse) {
+          this.errorHandler.handleError(error); // Envía el error al ErrorHandler global
+        }
+        return throwError(() => error);
+      })
+    );
+  }
+}
```

Comment:
Usemos el enfoque con `inject` que venimos usando. Está piola la solución del HTTP interceptor

```suggestion
@Injectable()
export class GlobalHttpErrorInterceptor implements HttpInterceptor {
  private _errorHandler = inject(ErrorHandler);

  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
    return next.handle(req).pipe(
      catchError((error) => {
        if (error instanceof HttpErrorResponse)
          this._errorHandler.handleError(error);
        return throwError(() => error);
      })
    );
  }
}
```